scripts/end-to-end/extract-testcases.py

```diff
@@ -0,0 +1,284 @@
+
+        for line in self.content.split("\n"):
+            line = line.strip()
+            if (not test_section) & line.startswith("//"):
+                comment += line[2:] + '\n'
+            if line.startswith("compileAndRun"):
+                test_section = True
+            if test_section & line.startswith("//"):
+                test_comment += line[2:]
+            if line.startswith("ABI_CHECK(") or line.startswith("BOOST_CHECK(") or line.startswith("BOOST_REQUIRE("):
+                comment = comment.strip()
+                if comment:
+                    self.comment = comment
+                test_comment = test_comment.strip()
+                self.tests.append(self.create_isoltest_call(line, test_comment))
+                comment = ""
+                test_comment = ""
+            else:
+                self.not_matching_lines.append(line)
+
+    def eval_and_correct(self, string):
+        if not string:
+            return ""
+        string = string.replace("u256(", "").replace(")", "").replace("string(", "").replace("encodeArgs(", "").replace(
+            "asString(", "").replace("fromHex(", "0x").replace("\"true\"", "true").replace("\"false\"", "false").strip()
+        result = ""
+        if string.startswith(","):
+            string = string[1:].strip()
+        if string.startswith("0x") and string.endswith("\""):
+            string = string.replace("\"", "")
+        if string:
+            r = ""
+            for arg in string.split(","):
+                arg = arg.strip()
+                if (arg == "true"):
+                    result += "true, "
+                elif (arg == "false"):
+                    result += "false, "
+                else:
+                    try:
+                        evaluated = eval(arg)
+                        if isinstance(evaluated, int):
+                            if string.find("0x") >= 0:
+                                r = hex(evaluated)
+                            else:
+                                r = int(evaluated)
+                        elif isinstance(evaluated, str):
+                            r = arg
+                        result += str(r) + ", "
+                    except (SyntaxError, NameError, TypeError, ZeroDivisionError):
+                        self.extractable = False
+                        result += "?, "
+            return result[:-2]
+        return ""
+
+    def create_call(self, signature, parameters, expectations, comment, withValue=False):
+        result = signature
+        value = ""
+        if withValue:
+            if parameters.find(",") == -1:
+                value = parameters
+                parameters = ''
+        if parameters:
+            result += ": "
+            result += parameters
+        if value:
+            result += ", " + str(value) + " wei"
+        result += " -> "
+        if expectations:
+            result += expectations
+        if comment:
+            result += " # " + comment + " #"
```

Comment:
Can you add the comment before the function call on a line of their own? This is how it was in SolidityEndToEndTests.cpp, right?

Reply:
yes, thats true.. sadly this is currently not supported by `isoltest`.. its only possible to comment on different parameters, or on the expectation

e.g. smoke/multiline_comments.sol
```
contract C {
    function f(uint a, uint b, uint c, uint d, uint e) public returns (uint) {
        return a + b + c + d + e;
    }
}
// ====
// compileViaYul: also
// ----
// f(uint256,uint256,uint256,uint256,uint256): 1, 1, 1, 1, 1
// # A comment on the function parameters. #
// -> 5
// f(uint256,uint256,uint256,uint256,uint256):
// 1,
// 1,
// 1,
// 1,
// 1
// -> 5
// # Should return sum of all parameters. #
```

Reply:
Please check `semanticTests/viaYul/detect_mul_overflow_signed.sol` - this looks like comments on the call to me.

Reply:
Or are you saying the comment is "after" an expectation?

Reply:
```

Reply:
We really have to change that. My impression is that comments refer to what follows.

Reply:
For example see structs/delete_struct.sol